docs/cluster_volumes.md

```diff
@@ -2,50 +2,97 @@ Cluster Volumes
-write storage plugins which are cross-compatible between various container
-orchestrators. However, most CSI plugins are shipped with configuration
-specific to Kubernetes. Docker CSI Plugins use the same binaries as those for
-Kubernetes, but in a different environment and sometimes with different
-configuration.
+Docker accesses CSI plugins through the Docker managed plugin system, using the
+`docker plugin` command.
 
-If a plugin is already adapted for and available for Docker, it can be
-installed through the `docker plugin install` command. Though such plugins may
-require configuration specific to the user's environment, they will ultimately
-be detected by and work automatically with Docker once enabled.
+If a plugin is available for Docker, it can be installed through the `docker
+plugin install` command. Plugins may require configuration specific to the
+user's environment, they will ultimately be detected by and work automatically
+with Docker once enabled.
 
 Currently, there is no way to automatically deploy a Docker Plugin across all
 nodes in a cluster. Therefore, users must ensure the Docker Plugin is installed
-on all nodes in the cluster on which it is desired.
+on all nodes in the cluster on which it is desired. 
+
+The CSI plugin must be installed on all manager nodes. If a manager node does
+not have the CSI plugin installed, a leadership change to that manager nodes
+will make Swarm unable to use that driver.
 
 Docker Swarm worker nodes report their active plugins to the Docker Swarm
-managers, and so it is not necessary to install a plugin on every worker node
-if this is not desired. However, the plugin must be installed on every manager
-node, or a leadership change could result in Docker Swarm no longer having the
-ability to call the plugin.
+managers, so it is not necessary to install a plugin on every worker node. The
+plugin only needs to be installed on those nodes need access to the volumes
+provided by that plugin.
+
+### Multiple Instances of the Same Plugin
+
+In some cases, it may be desirable to run multiple instances of the same
+plugin. For example, there may be two different instances of some storage
+provider which each need a differently configured plugin.
+
+To run more than one instance of the same plugin, set the `--alias` option when
+installing the plugin. This will cause the plugin to take a local name
+different from its original name.
 
-### Creating a Docker CSI Plugin
+Ensure that when using plugin name aliases, the plugin name alias is the same
+on every node.
+
+## Creating a Docker CSI Plugin
+
+Most CSI plugins are shipped with configuration specific to Kubernetes. They
+are often provided in the form of Helm charts, and installation information may
+include Kubernetes-specific steps. Docker CSI Plugins use the same binaries as
+those for Kubernetes, but in a different environment and sometimes with
+different configuration.
```

Comment:
I _think_ we need an example of how to configure a CSI plugin that would otherwise normally have configured that is Kubernetes-specific to run in Docker Swarm.

Can we pick a CSI plugin that's fairly simple, like maybe a CSI plugin for NFS or something non-cloud-specific and use that as an example to help support this improved documentation?

Reply:
I'd been hacking on the `democratic-csi` plugin to come up with these docs.

Reply:
Oh brilliant! 🙃